--- jquery-ui-autocomplete-1.10.2.js	2013-06-21 11:19:32.000000000 -0700
+++ jquery-ui-autocomplete-custom.js	2013-06-21 11:34:35.000000000 -0700
@@ -106,6 +106,7 @@
           suppressKeyPress = true;
           this._keyEvent( "next", event );
           break;
+        case keyCode.SPACE:
         case keyCode.ENTER:
         case keyCode.NUMPAD_ENTER:
           // when menu is open and has focus
@@ -121,6 +122,7 @@
           if ( this.menu.active ) {
             this.menu.select( event );
           }
+          n.preventDefault();
           break;
         case keyCode.ESCAPE:
           if ( this.menu.element.is( ":visible" ) ) {
